server/src/test/java/io/crate/replication/logical/action/PublicationsStateActionTest.java

```diff
@@ -73,9 +91,162 @@ public void test_resolve_relation_names_for_all_tables_ignores_table_with_soft_d
 
         var resolvedRelations = PublicationsStateAction.TransportAction.resolveRelationsNames(
             publication,
-            s.schemas()
+            s.schemas(),
+            user,
+            user
         );
         assertThat(resolvedRelations, contains(new RelationName("doc", "t1")));
         appender.assertAllExpectationsMatched();
     }
+
+    @Test
+    public void test_resolve_relation_names_for_all_tables_ignores_table_when_pub_owner_doesnt_have_read_write_define_permissions() throws Exception {
+        var publicationOwner = new User("publisher", Set.of(), Set.of(), null) {
+            @Override
+            public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, String ident, String defaultSchema) {
+                return (READ_WRITE_DEFINE.contains(type) && clazz.equals(Privilege.Clazz.TABLE) && ident.equals("doc.t1"));
+            }
+        };
+        var subscriber = new User("subscriber", Set.of(), Set.of(), null) {
+            @Override
+            public boolean hasPrivilege(Privilege.Type type, Privilege.Clazz clazz, String ident, String defaultSchema) {
+                return true;
+            }
+        };
+
+        UserLookup userLookup = mock(UserLookup.class);
+        when(userLookup.findUser("publisher")).thenReturn(publicationOwner);
+        when(userLookup.findUser("subscriber")).thenReturn(subscriber);
+
+        var s = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE doc.t1 (id int)")
+            .addTable("CREATE TABLE doc.t3 (id int)")
+            .build();
+        var publication = new Publication("publisher", true, List.of());
+
+        var resolvedRelations = PublicationsStateAction.TransportAction.resolveRelationsNames(
+            publication,
+            s.schemas(),
+            publicationOwner,
+            subscriber
+        );
+        assertThat(resolvedRelations, contains(new RelationName("doc", "t1")));
+        assertThat(resolvedRelations, not(contains(new RelationName("doc", "t3"))));
```

Comment:
`contains()` already will fail if `t3` is part of the collection, so not need for the `not(contains())` check